nixos/modules/services/system/dbus.nix

```diff
@@ -122,6 +141,30 @@ in
 
     systemd.services.dbus.restartTriggers = [ configDir ];
 
+    systemd.user = mkIf config.systemd.user.dbus.enable {
+      services.dbus = {
+        description = "D-Bus User Message Bus";
+        requires = [ "dbus.socket" ];
```

Comment:
I think it's because ```sockets.target``` is required so early in the process it's guaranteed to be up, but it does look like a race condition waiting to happen.